pkg/auth/providerrefresh/refresher.go

```diff
@@ -281,6 +270,27 @@ func (r *refresher) refreshAttributes(attribs *v3.UserAttribute) (*v3.UserAttrib
 			}
 		}
 
+		if principalID != "" && (len(loginTokens[providerName]) > 0 || (len(derivedTokens[providerName]) > 0 && canAccessProvider)) {
```

Comment:
I think I understand part of this if condition, but I'm not super sure I understand why we want to check `&& canAccessProvider`. It would seem to me that the below nil pointer issues can be prevented by the first part of this check.

I guess what I am asking is what value do we get out of the above as opposed to:

```go
if principalID != "" && (len(loginTokens[providerName]) > 0 || len(derivedTokens[providerName]) > 0)
```

I'm not saying we need to change it, just would like a better understanding of why the if statement is structured this way.

Reply:
Added a comment to explain.